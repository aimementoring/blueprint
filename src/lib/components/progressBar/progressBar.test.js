import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from './progressBar.js';

describe('ProgressBar', () => {
  it('renders properly with no props', () => {
    const tree = renderer.create(<ProgressBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with just progress value', () => {
    const tree = renderer.create(<ProgressBar progress={10} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with just progress value and display number on true', () => {
    const tree = renderer.create(<ProgressBar progress={40} displayNumber />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly using all available props', () => {
    const tree = renderer
      .create(
        <ProgressBar
          progress={45}
          displayNumber
          backgroundColor="green"
          barColor="red"
          color="yellow"
          height={40}
          fontSize={30}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
