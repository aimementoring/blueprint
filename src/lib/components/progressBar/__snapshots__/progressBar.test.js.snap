// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders properly even if the progress is > 100 1`] = `
<div
  className="theme-base"
>
  <div
    className="progressBar"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "fontSize": "undefinedpx",
          "lineHeight": "50px",
          "width": "100%",
        }
      }
    >
      100%
    </div>
  </div>
</div>
`;

exports[`ProgressBar renders properly using all available props 1`] = `
<div
  className="theme-base"
>
  <div
    className="progressBar"
    style={
      Object {
        "backgroundColor": "green",
        "height": "40px",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "backgroundColor": "red",
          "color": "yellow",
          "fontSize": "30px",
          "lineHeight": "40px",
          "width": "45%",
        }
      }
    >
      45%
    </div>
  </div>
</div>
`;

exports[`ProgressBar renders properly with just progress value 1`] = `
<div
  className="theme-base"
>
  <div
    className="progressBar"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "fontSize": "undefinedpx",
          "lineHeight": "50px",
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`ProgressBar renders properly with just progress value and display number on true 1`] = `
<div
  className="theme-base"
>
  <div
    className="progressBar"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "fontSize": "undefinedpx",
          "lineHeight": "50px",
          "width": "40%",
        }
      }
    >
      40%
    </div>
  </div>
</div>
`;

exports[`ProgressBar renders properly with no props 1`] = `
<div
  className="theme-base"
>
  <div
    className="progressBar"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "fontSize": "undefinedpx",
          "lineHeight": "50px",
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;
